dynamic_stack_decider/dynamic_stack_decider/sequence_element.py

```diff
@@ -16,21 +17,44 @@ class SequenceElement(AbstractStackElement):
     This is not an abstract class to inherit from.
     """
 
-    def __init__(self, blackboard, dsd: "DSD", actions: list[AbstractActionElement]):
+    def __init__(
+        self,
+        blackboard,
+        dsd: "DSD",
+        actions: list[ActionTreeElement],
+        init_function: Callable[[AbstractTreeElement], AbstractStackElement],
+    ):
         """
+        :param blackboard: Shared blackboard for data exchange and code reuse between elements
+        :param dsd: The stack decider which has this element on its stack.
         :param actions: list of initialized action elements
+        :param init_function: A function that initializes an action element creating a stack element from a tree element
         """
         super().__init__(blackboard, dsd, dict())
+        # Here we store the 'blueprints' of the actions
         self.actions = actions
-        self.current_action_index = 0
+        # We store a reference to the function that initializes the action elements based on the tree
+        self._init_function = init_function
+        # Here we store the actual instances of the active action
+        # The action is only initialized when it is the current action
+        assert len(actions) > 0, "A sequence element must contain at least one action"
+        self.current_action: AbstractActionElement = self._init_function(actions[0])
+        # Where we are in the sequence
+        self.current_action_index: int = 0
 
     def perform(self, reevaluate=False):
         """
         Perform the current action of the sequence. See AbstractStackElement.perform() for more information
 
         :param reevaluate: Ignored for SequenceElements
         """
+        # Log the active element
+        self.publish_debug_data("Active Element", self.current_action.name)
+        # Pass the perform call to the current action
```

Comment:
```suggestion
        # Call perform on the current action
```

Reply:
No we implement a perform call and pass that on to the currently active action.

Reply:
The other thing is obvious imo.

Reply:
I understand passing as the transfer of some data from some A to B. But you don't transfer anything around, you just call perform on B